contrib/src/main/java/com/datatorrent/contrib/parser/CellProcessorBuilder.java

```diff
@@ -0,0 +1,456 @@
+
+import com.datatorrent.contrib.parser.DelimitedSchema.FIELD_TYPE;
+
+/**
+ * Helper class with methods to generate CellProcessor objects. Cell processors
+ * are an integral part of reading and writing with Super CSV - they automate
+ * the data type conversions, and enforce constraints. They implement the chain
+ * of responsibility design pattern - each processor has a single, well-defined
+ * purpose and can be chained together with other processors to fully automate
+ * all of the required conversions and constraint validation for a single
+ * delimited record.
+ * 
+ */
+public class CellProcessorBuilder
+{
+
+  /**
+   * Method to get cell processors for given field type and constraints
+   * 
+   * @param fieldType
+   *          data type of the field
+   * @param constraints
+   *          a map of constraints
+   * @return
+   */
+  public static CellProcessor getCellProcessor(FIELD_TYPE fieldType, Map<String, Object> constraints)
+  {
+    switch (fieldType) {
+      case STRING:
+        return getStringCellProcessor(constraints);
+      case INTEGER:
+        return getIntegerCellProcessor(constraints);
+      case LONG:
+        return getLongCellProcessor(constraints);
+      case FLOAT:
+      case DOUBLE:
+        return getDoubleCellProcessor(constraints);
+      case CHARACTER:
+        return getCharCellProcessor(constraints);
+      case BOOLEAN:
+        return getBooleanCellProcessor(constraints);
+      case DATE:
+        return getDateCellProcessor(constraints);
+      default:
+        return null;
+    }
+  }
+
+  /**
+   * Method to get cellprocessor for String with constraints. These constraints
+   * are evaluated against the String field for which this cellprocessor is
+   * defined.
+   * 
+   * @param constraints
+   *          map of constraints applicable to String
+   * @return CellProcessor
+   */
+  private static CellProcessor getStringCellProcessor(Map<String, Object> constraints)
+  {
+    Boolean required = constraints.get(DelimitedSchema.REQUIRED) == null ? null : Boolean
+        .parseBoolean((String)constraints.get(DelimitedSchema.REQUIRED));
+    Integer strLen = constraints.get(DelimitedSchema.LENGTH) == null ? null : Integer.parseInt((String)constraints
+        .get(DelimitedSchema.LENGTH));
+    Integer minLength = constraints.get(DelimitedSchema.MIN_LENGTH) == null ? null : Integer
+        .parseInt((String)constraints.get(DelimitedSchema.MIN_LENGTH));
+    Integer maxLength = constraints.get(DelimitedSchema.MAX_LENGTH) == null ? null : Integer
+        .parseInt((String)constraints.get(DelimitedSchema.MAX_LENGTH));
+    String equals = constraints.get(DelimitedSchema.EQUALS) == null ? null : (String)constraints
+        .get(DelimitedSchema.EQUALS);
+    String pattern = constraints.get(DelimitedSchema.REGEX_PATTERN) == null ? null : (String)constraints
+        .get(DelimitedSchema.REGEX_PATTERN);
+
+    CellProcessor cellProcessor = null;
+    if (StringUtils.isNotBlank(equals)) {
```

Comment:
We can surely have multiple constraints. However for String datatype, the constraints we are supporting are  equals, regex, string length, min and max length. For these, specifying multiple wouldn't make much sense. So, even if user provide multiple constraints, e.g regex and equals OR equals and length, the if/else block adds appropriate processors.